nuxeo-features/nuxeo-annotation/nuxeo-annotation-api/src/main/java/org/nuxeo/ecm/annotation/AnnotationService.java

```diff
@@ -34,44 +34,53 @@
     /**
      * Creates an annotation on a document.
      *
-     * @param documentModel the document to annotate
+     * @param session the core session
+     * @param documentId the annotated document id
+     * @param documentXpath the annotated document Xpath
      * @param annotation the annotation to create
-     * @return the annotation id
+     * @return the created annotation
      */
-    String createAnnotation(DocumentModel documentModel, Annotation annotation);
+    Annotation createAnnotation(CoreSession session, String documentId, String documentXpath, Annotation annotation);
 
     /**
-     * Gets an annotation for a document.
+     * Gets an annotation.
      *
-     * @param documentModel the document
+     * @param session the core session
      * @param annotationId the annotation id
      * @return the annotation, or {@code null} if the annotation with the given id was not found
      */
-    Annotation getAnnotation(DocumentModel documentModel, String annotationId);
+    Annotation getAnnotation(CoreSession session, String documentId, String documentXpath, String annotationId);
```

Comment:
What's the reason for having the `documentId` and `documentXpath` parameters? They don't seem to be used in the basic implementation.

Reply:
I added it to keep consistency between this method and `getAnnotations(CoreSession session, String documentId, String documentXpath)`. 
But something that could be added in the basic implementation is to check if the fetched annotation is actually tied to the annotated document by checking `documentId` and `documentXpath` wdyt ?

Reply:
If the parameters aren't used I think they should be removed, IMHO.
About the consistency check, not sure if it's this method's responsibility but it's probably more of a high-level discussion: does it make sense to fetch an annotation no matter its target document or should it always be fetched for a given document?

Reply:
The `annotationId` is the annotation model id in our case, but we don't set any  constraint on what it could be (a number for example, where it makes sense to keep the id and path parameters for a different impl).
I don't think it's relevant to fetch an annotation without the annotated document info as it is always linked to it (even if not reflected in our implementation).

Reply:
Rename parameter to `xpath` as it's not the document's xpath (that doesn't mean anything), but the blob's xpath inside the document.

Reply:
:+1: